engine/framework/registry/factory.go

```diff
@@ -81,7 +87,15 @@ func (f *SimpleWorkerFactory[T, C]) DeserializeConfig(configBytes []byte) (Worke
 	var config C
 	config = reflect.New(reflect.TypeOf(config).Elem()).Interface().(C)
 	if err := json.Unmarshal(configBytes, config); err != nil {
-		return nil, errors.Trace(err)
+		return nil, errors.Annotate(errDeserializeConfig, err.Error())
```

Comment:
This does not seem to be the idiomatic way to use `Annotate`, which is typically used to provide additional context to an error, rather than provide cause.

Reply:
fixed, ptal